### PR TITLE
fix(profiling): clear finished thread ids from `ThreadSpanLinks`

### PR DESCRIPTION
**Pull request by taegyunkim**
## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Comment by P403n1x87** [comment]
Would it be possible for us to do something like https://github.com/P403n1x87/echion/blob/f1ce747d23a631ef48dade4012cd2dc6dc8f1a4d/echion/monkey/threading.py#L27? This way we could untrack (at least pure Python) threads as soon as they are done

**Comment by taegyunkim** [comment]
> Would it be possible for us to do something like [P403n1x87/echion@`f1ce747`/echion/monkey/threading.py#L27](https://github.com/P403n1x87/echion/blob/f1ce747d23a631ef48dade4012cd2dc6dc8f1a4d/echion/monkey/threading.py#L27)? This way we could untrack (at least pure Python) threads as soon as they are done

Oh yeah, we do have `stack_v2.unregister_thread()`, we can clear the mapping from there for the specific thread. Thanks for pointing that out!

**Review by P403n1x87 — APPROVED** [review]

File: releasenotes/notes/profiling-span-link-leak-e7eb1aca3362cd2e.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+fixes:
+  - |
+    profiling: when a Python thread finishes, frees memory used for mapping
```
Comment (emmettbutler) [inline releasenotes/notes/profiling-span-link-leak-e7eb1aca3362cd2e.yaml:4]:
```suggestion
    profiling: when a Python thread finishes, this change frees memory used for mapping
```

**Review by emmettbutler — APPROVED** [review]

**Comment by taegyunkim** [comment]
Added an integration test checking `unregister_thread()` is called when thread completes.

**Pull request merged**